workflow/controller/operator.go

```diff
@@ -2934,7 +2925,7 @@ func (woc *wfOperationCtx) computeMetrics(metricList []*wfv1.Prometheus, localSc
 				woc.reportMetricEmissionError("real time metrics can only be used with metric variables")
 				continue
 			}
-			value = strings.TrimSuffix(strings.TrimPrefix(value, "{{"), "}}")
+			value = strings.TrimSuffix(strings.TrimPrefix(value, "{{"), "}}") // TODO
```

Comment:
are you missing any TODO here?

Reply:
this should use ResolveVar, yes

Reply:
actually, not really feasible, code here is very bespoke to metrics